src/services/codefixes/importFixes.ts

```diff
@@ -0,0 +1,279 @@
+            const allSourceFiles = context.program.getSourceFiles();
+            const useCaseSensitiveFileNames = context.host.useCaseSensitiveFileNames ? context.host.useCaseSensitiveFileNames() : false;
+
+            const token = getTokenAtPosition(sourceFile, context.span.start);
+            const name = token.getText();
+            const allActions: CodeAction[] = [];
+            let allPotentialModules: Symbol[] = [];
+
+            const ambientModules = checker.getAmbientModules();
+            if (ambientModules) {
+                allPotentialModules = ambientModules;
+            }
+            for (const otherSourceFile of allSourceFiles) {
+                if (otherSourceFile !== sourceFile && otherSourceFile.symbol) {
+                    allPotentialModules.push(otherSourceFile.symbol);
+                }
+            }
+
+            for (const moduleSymbol of allPotentialModules) {
+                cancellationToken.throwIfCancellationRequested();
+
+                const exports = checker.getExportsOfModule(moduleSymbol) || [];
+                for (const exported of exports) {
+                    if (exported.name === name) {
+                        allActions.push(getCodeActionForImport(moduleSymbol));
+                    }
+                }
+            }
+
+            return allActions;
+
+            function getCodeActionForImport(moduleSymbol: Symbol): CodeAction {
+                // Check to see if there are already imports being made from this source in the current file
+                const existingDeclaration = forEach(sourceFile.imports, importModuleSpecifier => {
+                    const importSymbol = checker.getSymbolAtLocation(importModuleSpecifier);
+                    if (importSymbol === moduleSymbol) {
+                        return getImportDeclaration(importModuleSpecifier);
+                    }
+                });
+
+                if (existingDeclaration) {
+                    return getCodeActionForExistingImport(existingDeclaration);
+                }
+                else {
+                    return getCodeActionForNewImport();
+                }
+
+                function getImportDeclaration(moduleSpecifier: LiteralExpression) {
+                    let node: Node = moduleSpecifier;
+                    while (node) {
+                        if (node.kind !== SyntaxKind.ImportDeclaration) {
+                            node = node.parent;
+                        }
+
+                        return <ImportDeclaration>node;
+                    }
+                    return undefined;
+                }
+
+                function getCodeActionForExistingImport(declaration: ImportDeclaration): CodeAction {
+                    const moduleSpecifier = declaration.moduleSpecifier.getText();
+
+                    // We have to handle all of the different import declaration forms
+                    if (declaration.importClause) {

```

Comment:
we need to handle the case where the name of the exported symbol is "default" but the local is a different name. 

options:

```ts
import {default as Foo} from "mod";
```
or 
```ts
import default from "mod";
```